src/pal/src/include/pal/stackstring.hpp

```diff
@@ -129,14 +147,21 @@ class StackString
         return (const T *)m_buffer;
     }
 
+    operator const T * () const {  return GetString(); }
+
+    //Always preserves the existing content
     T * OpenStringBuffer(SIZE_T count)
     {
         Resize(count);
         return (T *)m_buffer;
     }
 
+    //count should not include the terminating null
     void CloseBuffer(SIZE_T count)
     {
+        if ( 0 > count)
```

Comment:
I dont' like this hiding of caller problem. There is no reason for the caller to pass in negative count. We should fix the callers instead. Moreover, SIZE_T is unsigned, so this check would never work.